sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java

```diff
@@ -340,13 +339,8 @@ public void populateDisplayData(DisplayData.Builder builder) {
      * Like {@link #to(String)}.
      */
     public Write<T> to(ValueProvider<String> outputPrefix) {
-      return toResource(NestedValueProvider.of(outputPrefix,
-          new SerializableFunction<String, ResourceId>() {
-            @Override
-            public ResourceId apply(String input) {
-              return FileBasedSink.convertToFileResourceIfPossible(input);
-            }
-          }));
+      return toResource(
+          NestedValueProvider.of(outputPrefix, FileBasedSink.convertToFileResourceFunction()));
```

Comment:
That was my first inclination. The reason I didn't is because this method in FileBasedSink seems to be a utility that may be used in multiple file-based IO transforms used with NestedValueProvider. As a result, it seemed like many places were going to need to wrap this method into a function, and by putting it here my hope is that we don't introduce the same problem (capturing the closure) in the next one.